### PR TITLE
[MAINT] use scm for version

### PR DESCRIPTION
**Pull request by Remi-Gau**
- setup hatch-vcs to use git to keep track of the version
- gitignore the file generated by hatch-vcs on install

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,6 +11,14 @@ __pycache__
 *.egg-info
 .coverage
 
+# documenation 
+doc
+doc_conf/auto_examples/
+doc_conf/generated/
+
+# file generated by hatch
```
Comment (Remi-Gau) [inline .gitignore:19]: this one will be update on every install of the package

**Review by bthirion — APPROVED** [review]
LGTM.

**Review by lionelkusch — DISMISSED** [review]
Thank you for the modifications.

I had modifications to access to the version if it's outside of release, as Nilearn. 
I think with it, the pull request will be complete.

**Review by lionelkusch — APPROVED** [review]
I include a test for the version.
The two additional cannot be tested due to its environment dependence, and there are simple.

**Comment by lionelkusch** [comment]
@bthirion

**Comment by bthirion** [comment]
Thx !

**Pull request merged**